### ISSUE TITLE
reduce amount of installed packages when activating `xserver.enable`

### ISSUE THREAD

**Issue by zseri** [NixOS/nixpkgs#146269]
I use wayland/sway for basically all desktop work, but the nixos module for my display manager (lightdm) requires that `xserver.enable` is set to `true`. This causes `environment.systemPackages` to be kinda polluted with packages like `xterm`, which I didn't even use when primarily using XMonad before.

It would be nice if the set of packages which are installed when `xserver.enable` is set to `true` could be split into a subset installed into `environment.systemPackages` and `environment.defaultPackages`. (so that if the user wants to reduce the installed packages to a more bare-minimum, they can just override `environment.defaultPackages = lib.mkForce [];`). This would require a change to the definition of `defaultPackages`, so that, instead of supplying a default value, it would just supply the set with "somewhat below normal" override priority, which would be needed to be replicated across all modules which would utilize this (currently there are afaik none besides `environment`).

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
  - environment
  - services.xserver.enable
  - services.xserver.displayManager.lightdm
```

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/issues/139984

**Comment by SuperSamus** [comment]
Partially solved by #168194.
